### DISCUSSION TITLE
How to fixed this <Provider> Error when testing my component to render

### DISCUSSION THREAD

**Discussion by SharveParihar** [callstack/react-native-testing-library#1582] [Q&A]
<img width="956" alt="Screenshot 2024-04-01 at 11 06 44 AM" src="https://github.com/callstack/react-native-testing-library/assets/154511396/9ada9bda-21d4-4062-810b-cb171583fcfd">

 PaperProvider  is already present in the code - 

<img width="871" alt="Screenshot 2024-04-01 at 11 10 06 AM" src="https://github.com/callstack/react-native-testing-library/assets/154511396/45cd5896-720a-4dd2-a790-55fdea9320df">

**Comment by guvenkaranfil** [comment]
Does `FABGroup` component contains the provider or upper level component provides it? If a upper component contains the provider ( like App.tsx ) and you test the below level component you should pass the provider when testing. You can accomplish this by wrapper property second parameter as [options](https://callstack.github.io/react-native-testing-library/docs/api#render-options) in render function

If it does not solve the problem let me know more detail about your implementation @SharveParihar

**Reply by SharveParihar** [reply]
I am using react-native-paper and provider is added at the root level of app, 

`
import {  PaperProvider  } from 'react-native-paper';

 < PaperProvider >  theme={DefaultTheme}
        // ... Other Code for FAB Group
      < /PaperProvider >
      
`

So I have also tried to add paper provider in my test file but it is failing with another error , that is given below - 

`test('if it renders correctly', () => {
    render(<Provider> <FABGroup {...FABGroup.args} /> </Provider>);

    // Check FAB Group is rendered
    const buttonElement = screen.getByTestId(FABGroup.args.testID);

    expect(buttonElement).not.toBeNull();
  });
`
Given below is the error - 

<img width="632" alt="Screenshot 2024-04-01 at 10 04 36 PM" src="https://github.com/callstack/react-native-testing-library/assets/154511396/cdf45675-a5c3-42f8-bb3a-1af0393347a2">
